editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx

```diff
@@ -0,0 +1,125 @@
+import { MyMap } from "@opensystemslab/map";
+import { Presentational as MapAndLabel } from "@planx/components/MapAndLabel/Public";
+import { waitFor } from "@testing-library/react";
+import React from "react";
+import { setup } from "testUtils";
+import { vi } from "vitest";
+import { axe } from "vitest-axe";
+
+import { point1 } from "../test/mocks/geojson";
+import { props } from "../test/mocks/Trees";
+import { addFeaturesToMap } from "../test/utils";
+
+beforeAll(() => {
+  if (!window.customElements.get("my-map")) {
+    window.customElements.define("my-map", MyMap);
+  }
```

Comment:
As opposed to calling this in multiple place (React app, Storybook, test suites), I think it's probably a better idea to make a basic wrapper for the map webcomponent (`<my-map/>`) which includes this in a `useEffect()` or similar.

Reply:
@jessicamcinchak @jamdelion Is this part of the test defining the `<my-map>` component as "my-map" within the ShadowDOM to make it retrievable?

Reply:
As I understand it - this checks if our custom element is already available/found in the window, and adds it if not ! As a custom web component, it naturally has a shadow DOM representation - we don't need to specifically add it to the shadow DOM, just access it from there inherently